src/json-patch.ts

```diff
@@ -89,8 +146,30 @@ export class JsonPatch {
    *
    * @example JsonPatch.test('/best_biscuit/name', 'Choco Leibniz')
    */
-  public static test(path: string, value: any) {
-    return new JsonPatch({ op: "test", path, value });
+  public static test(
+    path: string,
+    value: any,
+    failureBehavior: TestFailureBehavior = TestFailureBehavior.SKIP_PATCH
+  ) {
+    const patch = new JsonPatch({
+      op: "test",
+      path,
+      value,
+    } satisfies TestOperation<any>);
+    Object.defineProperty(patch, TEST_FAILURE_BEHAVIOR_SYMBOL, {
+      writable: false,
+      value: failureBehavior,
+    });
+    return patch;
+  }
+
+  /**
+   * Escapes a json pointer path
+   * @param path The raw pointer
+   * @return the Escaped path
+   */
+  public static escapePathComponent(path: string): string {
```

Comment:
This feels a bit less like leaking the underlying implementation through.

```suggestion
  public static escapePath(path: string): string {
```